packages/nimble-components/src/table/template.ts

```diff
@@ -48,11 +51,13 @@ export const template = html<Table>`
                     <div class="header-row" role="row">
                         <span class="header-row-action-container" ${ref('headerRowActionContainer')}>
                             ${when(x => x.selectionMode === TableRowSelectionMode.multiple, html<Table>`
-                                <span role="columnheader" class="checkbox-container">
+                                <span role="gridcell" class="checkbox-container">
```

Comment:
I'm curious if you have specific rationale for this change (other than that it's inconsistent with the role of the button below it).

I had a hard time trying to find guidance for the ARIA role of select-all buttons in a column header. The [AG Grid example](https://www.ag-grid.com/example/) still has role=columnheader on it's Select All Rows checkbox, but in their DOM structure, it's combined with the header text for the first column, so it's probably not a good example.

Reply:
I made this change because it fixed an accessibility problem and made the checkbox consistent with the "Collapse all" button below. That probably wasn't the best rationale.

I chatted with @atmgrifter00 about this since he's been looking at the impact of various roles on our elements within the table. We decided that this approach definitely seems wrong. From what we can tell, a correct approach is to put the checkbox + collapse button in a single `columnheader` with a non-visible label of something like "Row actions". That allows the row's selection checkbox (and the collapse button that will eventually come with hierarchy on some rows) to be in that column. As-is, the row selection checkbox is in a `gridcell` that causes the number of cells in a row to be unequal to the number of columns the grid claims to have.

Reply:
I've updated the structure/roles with these elements.